### PR TITLE
Dev/kduran/vbs22 65 agregar elemento presupuesto

### PR DESCRIPTION
**Pull request by kendallduranv**
Summary of the trip, also added multiple selection of the activities section

### CODE REVIEW

File: src/app/components/flight/flight-card/flight-card.component.ts [hunk 1]
```diff
@@ -65,27 +74,27 @@ export class FlightCardComponent {
       travel_class: this.initialForm.travel_class ?? 1,
       stops: this.initialForm.stops ?? 0
     };
-  
+
     // Validar 'type'
```
Comment (DieS06) [inline src/app/components/flight/flight-card/flight-card.component.ts:78]: Delete comment.

File: src/app/interfaces/flights.interface.ts [hunk 2]
```diff
@@ -1,25 +1,49 @@
...
 export interface Airport {  
-    name?                           :string;
-    id?                             :string;
-    time?                           :string;
+    name?                           : string;
+    id?                             : string;
+    time?                           : string;
 }
 
+
```
Comment (DieS06) [inline src/app/interfaces/flights.interface.ts:26]: Use interface with airport object.

File: src/app/interfaces/flights.interface.ts [hunk 3]
```diff
@@ -1,25 +1,49 @@
...
+export interface IVoiageFlight{
```
Comment (DieS06) [inline src/app/interfaces/flights.interface.ts:27]: Discuss as team why not to use the same object distribution with airport object inside flight as the database model.

File: src/app/interfaces/trip.interface.ts [hunk 4]
```diff
@@ -11,4 +19,30 @@ export interface ITripForm{
...
+    trip_name:          string;
+    trip_description:   string;
+    departure_city?:    string;
+    departure_date:     Date;
+    destination_city:   string;
+    return_date:        Date;
+    budget? :           number;
+    currency?:          ICurrency;
+    lodge:              IVoiageLodge; 
+    flight:             IVoiageFlight; // IFlights: ver lo de separarlo en departure y return flight
```
Comment (DieS06) [inline src/app/interfaces/trip.interface.ts:35]: Delete comment.

File: src/app/components/food/food-card/food-card.component.html [hunk 5]
```diff
@@ -15,8 +15,18 @@
 
                     <div class="align-boxes space-between">
                         <h1 class="bold-text font-size-title">{{item.name}}</h1>
-                        <h1 class="move-to-right font-size-price">Rating: {{item.rating}}</h1>
+
+                        <div class="form-check move-to-right ">
+                            <input class="form-check-input custom-checkbox " type="checkbox" value="" id="flexCheckDefault-{{item.id}}" (change)="checkboxChange(item, $event)">
```
Comment (DieS06) [inline src/app/components/food/food-card/food-card.component.html:20]: Indentation.

File: src/app/components/food/food-card/food-card.component.ts [hunk 6]
```diff
@@ -96,11 +111,42 @@ export class FoodCardComponent {
...
+      this.foodService.addItem(this.foodSelectedOption);
+      this.notifyService.onCustomSimpleNotify('Added to the list', 'The restaurant has been added to the list');
 
-    //Esto temporalmente mientras se completan demas componentes y borrar bien el local storage cuando se salga 
-    localStorage.removeItem('budget');
-    localStorage.removeItem('tripFormData');
+    } else {
+
+      this.foodSelectedOption.yelpId ? this.foodService.removeItem(this.foodSelectedOption.yelpId)  : 0;
+      this.notifyService.onCustomSimpleNotify('Removed from the list', 'The restaurant has been removed of the list');
```
Comment (DieS06) [inline src/app/components/food/food-card/food-card.component.ts:129]: Indentation.

File: src/app/components/lodge/lodge-card/lodge-card.component.html [hunk 7]
```diff
@@ -38,10 +38,10 @@ <h2>Amenities:</h2>
 
                     <div class="align-boxes space-between">
                         <button type="button" class="btn btn-outline-info ">View in map</button>
-                        <button type="button" class="btn btn-outline-dark" (click)="visitSite(item.link)">Go to the
+                        <button type="button" class="btn btn-outline-dark" (click)="visitSite(item.link)" >Go to the
```
Comment (DieS06) [inline src/app/components/lodge/lodge-card/lodge-card.component.html:41]: Indentation.

File: src/app/components/lodge/lodge-card/lodge-card.component.ts [hunk 8]
```diff
@@ -100,22 +108,48 @@ export class LodgeCardComponent {
...
+    this.lodgeSelected.lodge_name                    = googleLodge.name || '';
+    this.lodgeSelected.description                   = googleLodge.description || '';
+    this.lodgeSelected.check_in                      = this.initialForm.check_in_date || new Date();
+    this.lodgeSelected.check_out                     = this.initialForm.check_out_date || new Date();
+    this.lodgeSelected.night_price                   = googleLodge.rate_per_night?.extracted_lowest|| 0;
+    this.lodgeSelected.latitude                      = googleLodge.gps_coordinates?.latitude|| 0;
+    this.lodgeSelected.longitude                     = googleLodge.gps_coordinates?.longitude|| 0;
+    this.lodgeSelected.totalRate                     = googleLodge.overall_rating || 0;
+    this.lodgeSelected.external_link                 = googleLodge.link|| '';
+    this.lodgeSelected.images                        = googleLodge.images![0]?.original_image|| "./assets/img/No_image_available.png";
```
Comment (DieS06) [inline src/app/components/lodge/lodge-card/lodge-card.component.ts:143]: Indenation.

File: src/app/components/price-details/price-details.component.scss [hunk 9]
```diff
@@ -0,0 +1,83 @@
+@import url('https://fonts.googleapis.com/css2?family=Yeseva+One&display=swap');
```
Comment (DieS06) [inline src/app/components/price-details/price-details.component.scss:1]: No need to import, there is a link in index.html that laods the fonts to the project.

File: src/app/pages/trip-summary/trip-summary.component.scss [hunk 10]
```diff
@@ -0,0 +1,88 @@
+@import url('https://fonts.googleapis.com/css2?family=Yeseva+One&display=swap');
```
Comment (DieS06) [inline src/app/pages/trip-summary/trip-summary.component.scss:1]: No need to import fonts here, there is a link in the index.html that loads all the needed fonts.

File: src/app/services/voiage-services/trip.service.ts [hunk 11]
```diff
@@ -85,6 +98,35 @@ export class TripService extends BaseService<ITripForm> {
     } else {
       return this.onGetDefaultTripForm();
     }
-  }
+  };
+
+
+  //Valores por defecto para un trip
```
Comment (DieS06) [inline src/app/services/voiage-services/trip.service.ts:104]: Delete comment.

**Review by DieS06 — COMMENTED** [review]
.

**Review by mur-tz — APPROVED** [review]

**Review by nefi-cenfo — APPROVED** [review]

**Review by ichivancrCenfotec — APPROVED** [review]

File: src/app/interfaces/activities.interface.ts [hunk 12]
```diff
@@ -0,0 +1,7 @@
+export interface IActivities {
```
[the remark this answers is not in the document]
Comment (kendallduranv) [reply src/app/interfaces/activities.interface.ts:1]: This was provisional to later to adapt Ivan branch

**Comment by kendallduranv** [comment]
Merge

**Pull request merged**